powerauth-webflow-authentication-login-sca/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/method/loginsca/controller/LoginScaController.java

```diff
@@ -113,12 +124,26 @@ public LoginScaAuthResponse authenticateScaLogin(@Valid @RequestBody LoginScaAut
             String organizationId = request.getOrganizationId();
             AccountStatus accountStatus = statusConverter.fromUserAccountStatus(operation.getAccountStatus());
             boolean userIdAlreadyAvailable;
+            boolean userAuthenticatedUsingCertificate = false;
             if (userId == null) {
                 // First time invocation, user ID is not available yet
                 userIdAlreadyAvailable = false;
+                String clientCertificate = getClientCertificateFromHttpSession();
                 String username = request.getUsername();
-                ObjectResponse<UserDetailResponse> objectResponse = dataAdapterClient.lookupUser(username, organizationId, operationContext);
-                updateUsernameInHttpSession(username);
+                // Check that either certificate or username is available
+                if (clientCertificate == null && username == null) {
```

Comment:
Shouldn't be here `||`? Or rather changed the warning message which says `Missing username or client certificate`

Reply:
This was a typo in the log message, thanks for catching it.